src/main/java/com/conveyal/taui/models/ModificationStop.java

```diff
@@ -15,20 +15,41 @@
 
 class ModificationStop {
     private static double MIN_SPACING_PERCENTAGE = 0.25;
+    private static int DEFAULT_SEGMENT_SPEED = 15;
 
-    StopSpec stop;
-    double distanceFromStart;
+    private Coordinate coordinate;
+    private String id;
+    private double distanceFromStart;
 
-    ModificationStop(Coordinate c, String id, double distanceFromStart) {
-        this.stop = new StopSpec(c.x, c.y);
-        this.stop.id = id;
+    private ModificationStop(Coordinate c, String id, double distanceFromStart) {
+        this.coordinate = c;
+        this.id = id;
         this.distanceFromStart = distanceFromStart;
     }
 
+    /**
+     * Create the StopSpec types required by r5
+     * @param stops
+     * @return
+     */
     static List<StopSpec> toSpec (List<ModificationStop> stops) {
-        return stops.stream().map(s -> s.stop).collect(Collectors.toList());
+        return stops
+                .stream()
+                .map(s -> {
+                    if (s.id == null){
+                        return new StopSpec(s.coordinate.x, s.coordinate.y);
```

Comment:
Note for myself:

Using the different StopSpec constructors is the main fix for the bug.

Reply:
@ansoncfit I am adding comments to StopSpec to clarify this.